mathics/builtin/numeric.py

```diff
@@ -824,7 +825,437 @@ def apply(self, n, base, evaluation, nr_elements=None):
+    #> RealDigits[305.0123, 10, 17, 0]
+     = {{5, 0, 1, 2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, Indeterminate, Indeterminate, Indeterminate}, 1}
+     
+    #> RealDigits[220, 140]
+     = {{1, 80}, 2}
+     
+    #> RealDigits[Sqrt[3], 10, 50]
+     = {{1, 7, 3, 2, 0, 5, 0, 8, 0, 7, 5, 6, 8, 8, 7, 7, 2, 9, 3, 5, 2, 7, 4, 4, 6, 3, 4, 1, 5, 0, 5, 8, 7, 2, 3, 6, 6, 9, 4, 2, 8, 0, 5, 2, 5, 3, 8, 1, 0, 3}, 1}
+    
+    #> RealDigits[0]
+     = {{0}, 1}
+    
+    #> RealDigits[1]
+     = {{1}, 1}
+     
+    #> RealDigits[0, 10, 5]
+     = {{0, 0, 0, 0, 0}, 0}
+     
+    #> RealDigits[11/23]
+     = {{{4, 7, 8, 2, 6, 0, 8, 6, 9, 5, 6, 5, 2, 1, 7, 3, 9, 1, 3, 0, 4, 3}}, 0}
+    
+    #> RealDigits[1/97]
+     = {{{1, 0, 3, 0, 9, 2, 7, 8, 3, 5, 0, 5, 1, 5, 4, 6, 3, 9, 1, 7, 5, 2, 5, 7, 7, 3, 1, 9, 5, 8, 7, 6, 2, 8, 8, 6, 5, 9, 7, 9, 3, 8, 1, 4, 4, 3, 2, 9, 8, 9, 6, 9, 0, 7, 2, 1, 6, 4, 9, 4, 8, 4, 5, 3, 6, 0, 8, 2, 4, 7, 4, 2, 2, 6, 8, 0, 4, 1, 2, 3, 7, 1, 1, 3, 4, 0, 2, 0, 6, 1, 8, 5, 5, 6, 7, 0}}, -1}
+    
+    #> RealDigits[1/97, 2]
+     = {{{1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0}}, -6}
+   
+    #> RealDigits[1/197, 260, 5]
+     = {{1, 83, 38, 71, 69}, 0}
+     
+    #> RealDigits[1/197, 260, 5, -6]
+     = {{246, 208, 137, 67, 80}, -5}
+     
+    #> RealDigits[Pi, 260, 20]
+     = {{3, 36, 211, 172, 124, 173, 210, 42, 162, 76, 23, 206, 122, 187, 23, 245, 241, 225, 254, 98}, 1}
+  
+    #> RealDigits[Pi, 260, 5]
+     = {{3, 36, 211, 172, 124}, 1}
+     
+    #> RealDigits[1/3]
+     = {{{3}}, 0}
+     
+    #> RealDigits[1/2, 7]
+     = {{{3}}, 0}
+     
+    #> RealDigits[3/2, 7]
+     = {{1, {3}}, 1}
+    
+    #> RealDigits[-3/2, 7]
+     = {{1, {3}}, 1}
+     
+    #> RealDigits[3/2, 6]
+     = {{1, 3}, 1}
+     
+    #> RealDigits[1, 7, 5]
+     = {{1, 0, 0, 0, 0}, 1}
+     
+    #> RealDigits[I, 7]
+     : The value I is not a real number.
+     = RealDigits[I, 7]
+     
+    #> RealDigits[-Pi]
+     : The number of digits to return cannot be determined.
+     = RealDigits[-Pi]
+     
+    #> RealDigits[Round[x + y]]
+     = RealDigits[Round[x + y]]
+    
+    """
+    attributes = ('Listable',)
+    
+    messages = {
+        'realx': 'The value `1` is not a real number.',
+        'ndig': 'The number of digits to return cannot be determined.',
+        'rbase': 'Base `1` is not a real number greater than 1.',
+        'intnm': 'Non-negative machine-sized integer expected at position 3 in `1`.',
+        'intm' : 'Machine-sized integer expected at position 4 in `1`.',
+    }
+    
+    def apply_complex(self, n, var, evaluation):
+        'RealDigits[n_Complex, var___]'
+        return evaluation.message('RealDigits', 'realx', n)
+    
+    def apply_rational_with_base(self, n, b, evaluation):
+        'RealDigits[n_Rational, b_Integer]'
+        
+        expr = Expression('RealDigits', n)
+           
+        py_n = abs(n.value)
```

Comment:
better to use `get_int_value()`. `to_python` is when you don't know the type but need to coerce it to something python.